Add Makefile, configuration files and scripts

StefanJum opened the pull request:
Update the README with instructions on setting up, configuring, building and running the Python3 app with Unikraft. As part of it, add Makefile files, configuration files and scripts; they are referenced in the `README.md` file.

Current instructions cover the use of QEMU/KVM platform (both on x86_64 and on AArch64) and GCC. Firecraker/KVM, Xen, Linux platforms and Clang are not yet documented.

StefanJum committed:
Add Makefile, configuration files and scripts

Update the README with instructions on setting up, configuring, building
and running the Python3 app with Unikraft. As part of it, add Makefile
files, configuration files and scripts; they are referenced in the
`README.md` file.

Current instructions cover the use of QEMU/KVM platform (both on x86_64 and
on AArch64) and GCC. Firecraker/KVM, Xen, Linux platforms and Clang are
not yet documented.

Signed-off-by: Stefan Jumarea <stefanjumarea02@gmail.com>

John-Ted reviewed (approved):
Reviewed-by: Ioan-Teodor Teugea ioan_teodor.teugea@stud.acs.upb.ro

razvand commented on .gitignore:
```diff
@@ -0,0 +1,4 @@
+/.config
+/fs0/nginx/logs/
```
Copy-paste? For shame!

I would do a separate commit for ignoring output files.

razvand replied:
It should be `/fs0/`.

razvand commented on Makefile:
```diff
@@ -0,0 +1,10 @@
+
+$(MAKECMDGOALS):
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS) $(MAKECMDGOALS)
+
```
```suggestion
```

razvand commented on Makefile.uk:
```diff
@@ -0,0 +1,2 @@
+$(eval $(call addlib,apppython3))
+
```
```suggestion
```

razvand commented on README.md:
```diff
@@ -1,49 +1,305 @@
+* `git`
+* `wget`
+* `uuid-runtime`
+* `qemu-system-x86`
```
```suggestion
* `qemu-system-x86`
* `qemu-system-arm`
```

razvand commented on README.md:
```diff
@@ -1,49 +1,305 @@
+  wget \
+  uuid-runtime \
+  qemu-kvm \
+  qemu-system-x86 \
```
```suggestion
  qemu-system-x86 \
  qemu-system-arm \
```

razvand commented on README.md:
````diff
@@ -1,49 +1,305 @@
+  1. While inside the `python3/` directory, create the `fs0/` directory and extract the contents of `minrootfs.tgz`:
+
+     ```console
+     $ mkdir fs0/
+     $ cd fs0/
+     $ tar xvf ../minrootfs.tgz
+     $ cd ..
````
```suggestion
     $ mkdir fs0/
     $ tar -C fs0/ -xvf minrootfs.tgz
```

razvand reviewed (changes requested):
@StefanJum, see my comments. I would also do a commit to rename `minrootfs.tgz` to `rootfs.tar.gz`. The `.tgz` extension is weird. And no reason to confuse people about the `min` part in `minrootfs`.

StefanJum committed:
Add .gitignore file

Ignore files and folder that are output from the set up, configure,
build or run steps.

Signed-off-by: Stefan Jumarea <stefanjumarea02@gmail.com>

StefanJum committed:
Rename minrootfs.tgz to rootfs.tar.gz

Rename the filesystem archive to `rootfs.tar.gz`, since it is more
explicit.

Signed-off-by: Stefan Jumarea <stefanjumarea02@gmail.com>

razvand commented on README.md:
````diff
@@ -1,49 +1,305 @@
+On Ubuntu/Debian or other `apt`-based distributions, run the following command to install the requirements:
+
+```console
+$ apt-get install -y --no-install-recommends \
````
```suggestion
$ sudo apt install -y --no-install-recommends \
```

razvand commented on .gitignore:
```diff
@@ -0,0 +1,4 @@
+/.config
```
```suggestion
/.config
/.config.old
```

StefanJum committed:
kraft.yaml: Update platform name and rootfs path

Update platform name from `kvm` to `qemu`, since the default naming
convention was changed in release 0.13.0.
Also update the rootfs name, from `./minrootfs.tgz` to `./rootfs.tar.gz`.

Signed-off-by: Stefan Jumarea <stefanjumarea02@gmail.com>

razvand commented on README.md:
````diff
@@ -1,49 +1,305 @@
+  1. While inside the `python3/` directory, create the `fs0/` directory and extract the contents of `rootfs.tar.gz`:
+
+     ```console
+     $ mkdir fs0/
````
```suggestion
     $ rm -fr fs0/
     $ mkdir fs0/
```

razvand commented on README.md:
````diff
@@ -1,49 +1,305 @@
+
+     ```console
+     $ mkdir fs0/
+     $ tar -C fs0/ xvf ../rootfs.tar.gz
````
```suggestion
     $ tar -C fs0/ -xvf rootfs.tar.gz
```

razvand reviewed (changes requested):
@StefanJum, see my comments.

razvand reviewed (approved):
Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>
